src/SystemInterface.py

```diff
@@ -0,0 +1,101 @@
+        try:
+            cmd = [
+                "sudo",
+                "systemctl",
+                "status",
+                f"ggl.{component_name[0]}.service",
+            ]
+
+            # Run the command and stream output
+            process = subprocess.Popen(
+                cmd,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                universal_newlines=True,
+            )
+
+            output = process.stdout.read()
+            if output:
+                if len(output.split("\n")) > 2:
+                    if "Active: active (running)" in output.split(
+                            "\n")[2].strip():
+                        print(f"Process is active")
+                        return "RUNNING"
+
+            process.terminate()
+
+            return "NOT_RUNNING"
+
+        except KeyboardInterrupt:
+            print("\nStopping monitor...")
+            process.terminate()
+            return "NOT_RUNNING"
+        except Exception as e:
+            print(f"Error: {e}")
+            return "NOT_RUNNING"
+
+    def monitor_journalctl_for_message(self, service_name, message, timeout):
+        try:
+            cmd = [
+                "sudo",
+                "journalctl",
+                "-xeau",
+                service_name,
+                "-f",  # Follow mode - shows new entries as they are added
+            ]
+
+            # Run the command and stream output
+            process = subprocess.Popen(
+                cmd,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                universal_newlines=True,
+            )
+
+            print(f"Monitoring logs for {service_name}...")
+            timeout = time.time() + timeout
+
+            # Call the readline is blocking, set it to non-blocking mode.
+            os.set_blocking(process.stdout.fileno(), False)
+
+            while True:
+                # Check timeout
+                if time.time() > timeout:
+                    print(f"Timeout after {timeout} seconds")
+                    process.terminate()
+                    return False
+
+                output = process.stdout.readline()
+                if output:
+                    if message in output.strip():
+                        print(f"Found log")
+                        return True
+
+                # Check if process has terminated
+                if process.poll() is not None:
+                    print("journalctl process killed.")
+                    return False
+
+                time.sleep(0.01)
+
+        except KeyboardInterrupt:
+            print("\nStopping monitor...")
+            process.terminate()
+            return False
+        except Exception as e:
+            print(f"Error: {e}")
+            return False
+        finally:
+            # Ensure process is terminated
+            try:
+                process.terminate()
+            except:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are generically passing an `Exception` object without performing any other operation on it. This may hide error conditions that can otherwise be quickly detected and addressed. We recommend that you catch a more specific exception. If the code must broadly catch all exceptions, consider logging the stack trace using the [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) API. For example,

```
try:
    x = 1 / 0
except ZeroDivisionError as e:
    logging.exception(&#39;ZeroDivisionError: %s&#39;, e)

```